.github/workflows/build-test.yaml

```diff
@@ -3389,6 +3389,304 @@ jobs:
+          kubectl create secret docker-registry kotsadm-dockerhub --docker-server index.docker.io --docker-username "${{ secrets.E2E_DOCKERHUB_USERNAME }}" --docker-password "${{ secrets.E2E_DOCKERHUB_PASSWORD }}" --namespace "$KOTS_NAMESPACE"
+
+      - name: run upgrade-to-replicated-sdk test
+        id: upgrade-to-replicated-sdk
+        env:
+          APP_SLUG: upgrade-to-replicated-sdk
+        run: |
+          set +e
+
+          echo ${{ secrets.UPGRADE_TO_REPLICATED_SDK_LICENSE }} | base64 -d > license.yaml
+          LICENSE_ID="$(grep -oP 'licenseID:\s*\K\w+' license.yaml)"
+          echo "license-id=$LICENSE_ID" >> "$GITHUB_OUTPUT"
+
+          ./bin/kots \
+            install "$APP_SLUG/automated" \
+            --license-file license.yaml \
+            --no-port-forward \
+            --namespace "$KOTS_NAMESPACE" \
+            --shared-password password \
+            --app-version-label v1.0.0 \
+            --kotsadm-registry ttl.sh \
+            --kotsadm-namespace automated-${{ github.run_id }} \
+            --kotsadm-tag 24h
+
+          EXIT_CODE=$?
+          if [ $EXIT_CODE -ne 0 ]; then
+            echo "------pods:"
+            kubectl -n "$KOTS_NAMESPACE" get pods
+            echo "------kotsadm logs"
+            kubectl logs -l app=kotsadm --tail=100 --namespace "$KOTS_NAMESPACE"
+            exit $EXIT_CODE
+          fi
+
+          # wait for the app to be ready
+          COUNTER=1
+          while [ "$(./bin/kots get apps --namespace "$KOTS_NAMESPACE" | awk 'NR>1{print $2}')" != "ready" ]; do
+            ((COUNTER += 1))
+            if [ $COUNTER -gt 120 ]; then
+              echo "Timed out waiting for app to be ready"
+              ./bin/kots get apps --namespace "$KOTS_NAMESPACE"
+              echo "kotsadm logs:"
+              kubectl logs -l app=kotsadm --tail=100 --namespace "$KOTS_NAMESPACE"
+              exit 1
+            fi
+            sleep 1
+          done
+
+          # upgrade the app to the new version
+          ./bin/kots upstream upgrade "$APP_SLUG" -n "$KOTS_NAMESPACE" --deploy
+          sleep 5
+
+          # wait for the app to be ready again
+          COUNTER=1
+          while [ "$(./bin/kots get apps --namespace "$KOTS_NAMESPACE" | awk 'NR>1{print $2}')" != "ready" ]; do
+            ((COUNTER += 1))
+            if [ $COUNTER -gt 120 ]; then
+              echo "Timed out waiting for app to be ready"
+              ./bin/kots get apps --namespace "$KOTS_NAMESPACE"
+              echo "kotsadm logs:"
+              kubectl logs -l app=kotsadm --tail=100 --namespace "$KOTS_NAMESPACE"
+              exit 1
+            fi
+            sleep 1
+          done
+
+      - run: rm -rf ./replicated-sdk
+      - name: Checkout replicated-sdk
+        uses: actions/checkout@v3
+        with:
+          repository: replicatedhq/replicated-sdk
+          path: replicated-sdk
+          ref: main # TODO: make this variable based on version of SDK running in the cluster
```

Comment:
We need a tag of the SDK that contains the `namespace` input in order to get this working dynamically.